### ISSUE TITLE
Fusion Model Design + Hugging Face Integration for Backend Inference

### ISSUE THREAD

**Issue by lukhsaankumar** [McMasterAI-Society/DeepFakeDetector#48]
## Issue Type

- [ ] **Model**: ML model bug, training issue, or architecture problem
- [ ] **Data**: Dataset issue, preprocessing bug, or data pipeline problem  
- [ ] **Web**: Frontend bug or UI issue in the Next.js dashboard
- [ ] **API**: Backend API bug or FastAPI endpoint issue
- [x] **Research**: Research question or experimental feature request
- [ ] **Documentation**: Documentation bug or improvement needed
- [ ] **Bug**: General bug fix needed
- [ ] **Enhancement**: New feature or improvement request

## Description
Research, design, and document the **fusion (stacking) model** used to combine outputs from multiple submodels (CNN Transfer, ViT Base, DeiT Distilled, Gradient Field CNN) into a single final prediction: **"AI-generated vs Real" with confidence**.

The output of this ticket is a **clear, backend-ready design document** explaining:
- What fusion model to use
- How it is trained
- How it is stored on Hugging Face
- How the backend pulls it
- How inference flows end-to-end

This research document will be the blueprint for implementing the `/predict` and `/explain` backend endpoints.

## Acceptance Criteria
- [ ] Research document created at `research/fusion_model_design.md`
- [ ] Document covers all 7 required sections (see below)
- [ ] Document provides clear, actionable technical design
- [ ] Fusion model choice is justified with comparative analysis
- [ ] Training workflow is documented step-by-step
- [ ] Hugging Face storage format is clearly specified
- [ ] End-to-end inference flow is documented as execution trace
- [ ] Explainability approach is outlined

### Primary Deliverable
Create a markdown file: `research/fusion_model_design.md`

This file must clearly and concretely outline everything needed so that:
- The fusion model can be trained
- Uploaded to Hugging Face
- Pulled by the backend
- Run deterministically during inference

### Document Scope
The research document MUST cover the following 7 sections:

### 1) Fusion Model Choice (What Model & Why)
Research and justify **which fusion model should be used**, considering:
- Simplicity
- Stability
- Explainability
- Speed
- Suitability for conference demo + paper

**Required comparison** of at minimum:
- Logistic Regression (stacking on probabilities)
- Small MLP (2–3 layers)
- Rule-based weighted average (baseline)

**Clearly recommend ONE primary fusion model** (expected: Logistic Regression stacking) and explain:
- Why it is the best default
- When/if an upgrade would be justified

### 2) Fusion Model Inputs & Outputs
Define **exact input feature vector** to the fusion model, e.g.:
```
x = [
    p_cnn_transfer,
    p_vit_base,
    p_deit_distilled,
    p_gradfield_cnn
]
```

Optionally discuss:
- Entropy / confidence features
- Disagreement metrics (std of probabilities)

Define **output**:
- `prob_fake ∈ [0,1]`
- Final label via threshold

### 3) Training Workflow for the Fusion Model
Explain how the fusion model is trained:
- What dataset is used (validation split or held-out set)
- How submodel predictions are generated and cached
- How labels are aligned
- How overfitting is avoided
- What metrics are used to evaluate fusion performance

This should be understandable without seeing code.

### 4) How the Fusion Model is Stored on Hugging Face
Specify **exactly** how the fusion model is packaged on Hugging Face.

Expected structure:
```
DeepFakeDetector/fusion-logreg/
├── fusion.pkl          # sklearn or torch model
├── config.json         # input features, threshold, notes
├── label_map.json
└── README.md
```

Document:
- File formats
- What framework is used (sklearn / torch)
- Why this format is backend-friendly

### 5) How the Backend Pulls the Fusion Model from Hugging Face
Explain the backend flow conceptually:
- When the fusion model is downloaded (startup vs lazy load)
- How it is cached
- How versions are pinned (tag / commit hash)

Example (conceptual, not code):
```
snapshot_download("DeepFakeDetector/fusion-logreg")
load fusion.pkl into memory
```

### 6) End-to-End Inference Flow (CRITICAL)
Clearly outline the **full inference pipeline**:

1. Backend receives image
2. Image is preprocessed
3. Image is passed to each submodel:
   - CNN Transfer
   - ViT Base
   - DeiT Distilled
   - Gradient Field CNN
4. Each submodel outputs `prob_fake`
5. Fusion model receives `[p1, p2, p3, p4]`
6. Fusion model outputs:
   - Final `prob_fake`
   - Final label
7. Backend returns:
   - Overall AI-generated percentage
   - Per-submodel probabilities (optional)
   - Fusion confidence

**This section should read like a step-by-step execution trace.**

### 7) Explainability at the Fusion Level
Explain (conceptually) how fusion explainability is handled:
- Fusion model weights (importance of each submodel)
- How to expose this to the frontend
- How this complements Grad-CAM / attention maps

Example:
```
fusion weights:
  cnn_transfer: 0.32
  deit_distilled: 0.41
  vit_base: 0.18
  gradfield_cnn: 0.09
```

---

**Comment by Vihaan-Singhal1** [comment]
Completed in PR #50 (merged Feb 10, 2026). All 7 required sections are present and actionable in `research/fusion_model_design.md`.
